movie_search/media.py

```diff
@@ -1,113 +1,265 @@
-from movie_search import tmdb_api
-from .models import Movie, Video, Genre, Provider, Recommendation
+from abc import ABC, abstractmethod
+from .models import (
+    Movie,
+    MovieVideo,
+    MovieGenre,
+    MovieProvider,
+    MovieRecommendation,
+    TVSeries,
+    TVSeriesGenre,
+    TVSeriesProvider,
+    TVSeriesRecommendation,
+)
+from .tmdb_api import TMDBApi
 
+tmdb_api_obj = TMDBApi()
 
 
+# standalone functions for general fetching of data
 def fetch_data_from_api(endpoint, **kwargs):
     """A common method to fetch data from media API"""
-    return tmdb_api.get_data_from_endpoint(endpoint, **kwargs)
+    return tmdb_api_obj.get_data_from_endpoint(endpoint, **kwargs)
+
 
 def fetch_api_data_by_query(endpoint, text_query, results_key):
     """A common method to fetch data by query from media API"""
-    return tmdb_api.get_data_by_query(endpoint, text_query, results_key)
+    return tmdb_api_obj.get_data_by_query(endpoint, text_query, results_key)
+
 
 def lookup_id_in_data_by_query(data_dict, query):
     """A common method to fetch media ID from data by query"""
     return {item.lower(): idx for item, idx in data_dict.items()}.get(query)
 
 
-def fetch_and_store_movie_from_api(movie_id):
-    movie_data = tmdb_api.get_data_from_endpoint(f"/movie/{movie_id}")
-    movie = store_movie_data(movie_data)
-    videos = store_movie_videos(movie)
-    return movie, videos
-
-def store_movie_data(data):
-    genres = data.get("genres")
-    movie, created = Movie.objects.get_or_create(
-        movie_id=data.get("id", 0),
-        title=data.get("title", ""),
-        backdrop_path=data.get("backdrop_path", ""),
-        tagline=data.get("tagline", ""),
-        vote_count=data.get("vote_count", 0),
-        vote_average=data.get("vote_average", 0),
-        popularity=data.get("popularity", 0),
-        release_date=data.get("release_date", ""),
-        runtime=data.get("runtime", 0),
-        production_company=data.get("production_company", ""),
-        overview=data.get("overview", ""),
-        budget=data.get("budget", 0),
-        revenue=data.get("revenue", 0),
-        homepage=data.get("homepage", ""),
-    )
-    if genres is not None:
-        movie_genres = store_movie_genres(genres)
-    
-    movie.genres.add(*movie_genres)
-    
-    movie_recommendations = store_movie_recommendations(movie.movie_id)
-    movie.recommendation.add(*movie_recommendations)
+class MediaService(ABC):
```

Comment:
TIL `class PybitesSearch(metaclass=ABCMeta):` -> I can define it as `class PybitesSearch(ABC)` - should work the same.

Reply:
Cool, thanks